dipy/denoise/localpca.py

```diff
@@ -10,39 +10,86 @@
 from scipy.linalg import eigh
 
 
-def localpca(arr, sigma, mask=None, pca_method='eig', patch_radius=2,
-             tau_factor=2.3, out_dtype=None):
+def _pca_classifier(L, m):
+    """ Classifies which PCA eigenvalues are related to noise and estimates the
+    noise variance
+
+    Parameters
+    ----------
+    L : array (n,)
+        Array containing the PCA eigenvalues in ascending order.
+
+    Returns
+    -------
+    var : float
+        Estimation of the noise variance
+    ncomps : int
+        Number of eigenvalues related to noise
+
+    Notes
+    -----
+    This is based on the algorithm described in [1]_.
+
+    References
+    ----------
+    .. [1] Veraart J, Novikov DS, Christiaens D, Ades-aron B, Sijbers,
+           Fieremans E, 2016. Denoising of Diffusion MRI using random matrix
+           theory. Neuroimage 142:394-406.
+           doi: 10.1016/j.neuroimage.2016.08.016
+    """
+    var = np.mean(L)
+    c = L.size - 1
+    r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / m) * var
+    while r > 0:
+        var = np.mean(L[:c])
+        c = c - 1
+        r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / m) * var
+    ncomps = c + 1
+    return var, ncomps
+
+
+def localpca(arr, sigma=None, mask=None, pca_method='eig', patch_radius=2,
```

Comment:
If we move the `_noise_classifier` function to `noise_estimate.py` then, we can have an option here that says `method='lpca'` or `method='mppca'` ? This will also help if we need to add any other noise estimation methods in the future.. Does this make sense? Your current design is good and should be able to support this capability!

Reply:
That would make sense. However, the _pca_classifier is just an auxiliar function for mppca. The noise estimate from mppca requires the full localpca body, so having the noise estimation from these method on the noise_estimate.py file will imply having to duplicate the full localpca code there.

Reply:
I understand! Sounds good! Resolving this 👍